### ISSUE TITLE
Upload goes beyond 100%

### ISSUE THREAD

**Issue by neilyoung** [gotev/android-upload-service#161]
With the update to 3.0 OkHTTP "large" uploads go beyond 100 %.

I'm using this setup

                new MultipartUploadRequest(mContext, uploadID, serverUrlString)
                        .setBasicAuth(username, password)
                        .addFileToUpload(videoFile, "VIDEO")
                        .addParameter("PANORAMA", ""+panorama)
                        .setMaxRetries(1)
                        .setDelegate(videoUploadResult)
                        .startUpload();

The traces  come from onProgress. 

       public void onProgress(UploadInfo uploadInfo) {
            Log.i(TAG, "Upload ID "
                    + uploadInfo.getUploadId() + " progress: " + uploadInfo.getProgressPercent());
            Log.i(TAG, "Upload ID " + uploadInfo.getUploadId() + " uploaded bytes: " + uploadInfo.getUploadedBytes() + ", total: " + uploadInfo.getTotalBytes());
            mProgressbar.setProgress(uploadInfo.getProgressPercent());
       }

    08-01 12:21:55.191 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d uploaded bytes: 12878397, total: 13384755
    08-01 12:21:55.412 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d progress: 97
    08-01 12:21:55.412 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d uploaded bytes: 13099581, total: 13384755
    08-01 12:21:55.646 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d progress: 99
    08-01 12:21:55.646 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d uploaded bytes: 13328957, total: 13384755
    08-01 12:22:26.930 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d progress: 100
    08-01 12:22:26.930 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d uploaded bytes: 13384804, total: 13384755
    08-01 12:22:27.478 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d progress: 102
    08-01 12:22:27.478 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d uploaded bytes: 13712960, total: 13384755
    08-01 12:22:27.658 ImyApp( 8386): Upload ID 59a7a6c5-b0bd-4a67-a380-b93f8033686d progress: 103

It seems onCompleted is never called

**Comment by neilyoung** [comment]
Simple enough: I had the retry count to 1 and the server did hung the request for > 30 secs. This seems to be a case you are not handling correctly. The server retries, but the counters are not reset before.

**Comment by gotev** [comment]
Yes, you can. Just create your own instance of the OkHttpClient, set the desired timeout and then pass it to the OkHttpStack constructor when you initialize the library

**Comment by gotev** [comment]
@neilyoung @AngleV @konalegi just released 3.0.1 which updates OkHttp stack from 3.2.0 to 3.4.1. Please test it and tell me if this problem is still present.

**Comment by AngleV** [comment]
@gotev thanks

will perform some tests and i'll get back to you !!!

**Comment by AngleV** [comment]
@gotev 
i am getting the following warn when i try to compile 

```
Warning: okhttp3.internal.huc.OkHttpsURLConnection: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
```

**Comment by hendrawd** [comment]
This happens to HurlStack too, and the file uploaded sometimes twice or triple, but actually only upload once

**Comment by gotev** [comment]
@hendrawd don't know your server implementation, but if something goes wrong with HTTP/Multipart before the upload gets completed successfully, it's up to the server to discard the partially uploaded files. So can you confirm that by introducing a 30s delay in the server response and setMaxRetries to 1 by using the demo server written in node.js I can reproduce the error?

@AngleV I've looked at my build log and I don't have that warning message. The same applies also to the automatic build logs on Travis CI. Check if you have used different OkHttp versions in your app in your dependencies

**Comment by hendrawd** [comment]
@gotev Actually it is not me who maintained code on server, so i don't know exactly how the server will handle, but the first upload is not partially uploaded files. All the same files uploaded fully to server.

**Comment by gotev** [comment]
What I need are precise steps to reproduce the error by using the demo app and the demo server provided in the examples, otherwise I can't address the issue

**Comment by gotev** [comment]
@neilyoung @hendrawd @AngleV just released 3.0.3 which contains the fix to this issue. It should have been solved now. Try and let me know :)
